service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcService.java

```diff
@@ -0,0 +1,35 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.storage.StorageConnectorService;
+import uk.nhs.adaptors.gp2gp.ehr.request.EhrExtractRequestHandler;
+
+@Service
+@Slf4j
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class GpcService {
+
+    private final GpcClient gpcClient;
+    private final GpcRequestBuilder gpcRequestBuilder;
+    private final StorageConnectorService storageConnectorService;
+    private final EhrExtractRequestHandler ehrExtractRequestHandler;
+
+    public void handleStructureTask(GetGpcStructuredTaskDefinition structuredTaskDefinition) throws IOException {
+        var ehrExtractStatus = ehrExtractRequestHandler.getEhrExtractStatus(structuredTaskDefinition);
```

Comment:
Instead of fetching the ehrExtractStatus could we include toAsid and fromAsid in task definition? Avoid making a database request that is not needed